src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -31,76 +39,167 @@ public String greeting(@RequestParam(name="name", required=false, defaultValue="
+    @ApiOperation(value = "Deletes a book by id.", response = void.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "OK"),
+        @ApiResponse(code = 404, message = "Book not found for ID.")
+    })
     @DeleteMapping("/{id}")
-    public void deleteById(@PathVariable final Long id) {
+    public void deleteById(
+        @ApiParam(value = "Book to delete id", required = true)
+        @PathVariable final Long id) {
         bookRepository.findById(id)
             .orElseThrow(() -> new BookNotFoundException(id));
         bookRepository.deleteById(id);
     }
 
+    @ApiOperation(value = "Deletes all books written by some author.", response = void.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "OK"),
+        @ApiResponse(code = 404, message = "Book not found for the author."),
+        @ApiResponse(code = 400, message = "Book could not be deleted.")
+    })
     @DeleteMapping("/author/{author}")
-    public void deleteAllByAuthor(@PathVariable final String author) {
-        bookRepository.findOneByAuthor(author)
-            .orElseThrow(() -> new AuthorDoesNotHaveBooksException(author));
-        bookRepository.deleteByAuthor(author);
+    public void deleteAllByAuthor(
+        @ApiParam(value = "author name to delete the books", required = true)
+        @PathVariable final String author) {
+        try {
+            List<Book> booksToDelete = bookRepository.findByAuthor(author);
+            if (booksToDelete.isEmpty()) {
+                throw new AuthorDoesNotHaveBooksException(author);
+            }
+            for (Book book : booksToDelete) {
+                bookRepository.deleteById(book.getId());
+            }
+        } catch (Exception e) {
+            throw new FailedToDeleteBookException(e);
+        }
     }
 
+    @ApiOperation(value = "Deletes all books from a publisher.", response = void.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "OK"),
+        @ApiResponse(code = 404, message = "Book not found for the publisher."),
+        @ApiResponse(code = 400, message = "Book could not be deleted.")
+    })
     @DeleteMapping("/publisher/{publisher}")
-    public void deleteAllByPublisher(@PathVariable final String publisher) {
-        bookRepository.findOneByPublisher(publisher)
-            .orElseThrow(() -> new PublisherDoesNotHaveBooksException(publisher));
-        bookRepository.deleteByPublisher(publisher);
+    public void deleteAllByPublisher(
+        @ApiParam(value = "Publisher to delete the books", required = true)
+        @PathVariable final String publisher) {
+        try {
+            List<Book> booksToDelete = bookRepository.findByPublisher(publisher);
+            if (booksToDelete.isEmpty()) {
+                throw new PublisherDoesNotHaveBooksException(publisher);
+            }
+            for (Book book : booksToDelete) {
+                bookRepository.deleteById(book.getId());
+            }
```

Comment:
A little comment :)
Another way to do this is use a reposotory method "deleteAllByPublisher".
"deleteAll" equals to "findAll().forEach(this::delete)".